torch/_subclasses/fake_tensor.py

```diff
@@ -1511,20 +1542,20 @@ def validate_and_convert_non_fake_tensors(self, func, converter, args, kwargs):
 
         def validate(x):
             nonlocal flat_arg_fake_tensors
-            if not isinstance(x, FakeTensor):
+            if not self.is_our_fake(x):
                 if torch.Tag.inplace_view in func.tags:  # type: ignore[attr-defined]
                     raise Exception(
                         f"Can't call metadata mutating ops on non-Fake Tensor inputs. Found in {render_call(func, args, kwargs)}"
                     )
                 if not self.allow_non_fake_inputs:
+                    if x.fake_mode is not self:
```

Comment:
Out of curiosity, 

1) when we `allow_non_fake_inputs`, does it mean dynamo will actually perform computation on the input tensors or it means it will be fakefied somewhere else, before tracing?

2) when we don't allow, is it ok to have part of the input fakefied with user (outer) fake mode and some other converted here using the dynamo (inner) fake mode?

3) Where in the code the model parameters are fakefied?

Reply:
1. It refakeifies the already fake tensors. This is required because otherwise you can't setup the symbolic sizes
2. Yes, in particular, the parameters of your model will be the outer fake, but the fake tensors in export passed to make_fx are going to be the inner fake
3. In Dynamo, model parameters get fakeified when we wrap the tensor. But they never get fakeified in the make_fx call; we rely on allow_non_fake_inputs to allow for this limited mixing